### PR TITLE
feat: add ML-based surrogate modeling module to experimental

### LINKED ISSUE #3169: ML-Based Surrogate Modeling Module for Efficient Parameter Exploration
**What's the problem this feature will solve?**

Analyzing Agent-Based Models (ABMs) often requires performing extensive Monte Carlo simulations to understand how different parameters impact model outcomes. This process is typically handled via the `BatchRunner` and is computationally expensive and time consuming, especially for complex simulations like the Sugarscape or Wolf-Sheep Predation models. Currently, there is no native way within the framework to approximate model behavior without running the full simulation every time, which creates a significant bottleneck for researchers conducting large-scale sensitivity analysis.

**Describe the solution you'd like**

I propose the implementation of a Surrogate Modeling module that allows users to train statistical emulators to replicate their model's behavior. Key features would include:

1) A pipeline to feed data directly from the `DataCollector` into `Scikit-learn` to fit a surrogate model.

2) Support for efficient parameter space sampling techniques, such as Latin Hypercube Sampling, to minimize the number of full model runs required for training.

3) A simple API to query the trained surrogate for predicted model outcomes at any point in the parameter space, providing near instant results for exploration.

4) Built-in tools to compare surrogate predictions against actual simulation results to verify the emulator's reliability.

**Additional context**

This module would leverage the existing structure of the `Model` and `AgentSet` classes to efficiently gather training data. By bridging the gap between Mesa and standard machine learning libraries, this feature would enable users to explore high-dimensional parameter spaces that are currently too computationally expensive to simulate fully. I have experience contributing to major machine learning repositories and would like to lead the development of this integration.

### PR DESCRIPTION
**Pull request by Nithurshen**
Fixes #3169 

### Summary

Introduces a new experimental package, `mesa.experimental.surrogate`, which enables the creation of statistical emulators (surrogates) for Mesa models. This module provides a streamlined pipeline to sample parameter spaces efficiently, run focused simulations, and train machine learning regressors to approximate model outcomes.

### Motive

Computational overhead is a significant bottleneck when performing large-scale sensitivity analysis or parameter exploration on complex Agent-Based Models (ABMs). Currently, researchers must execute a full simulation for every coordinate in a parameter space. This feature addresses the problem by allowing users to train a "proxy" model on a sparse set of data, enabling near-instant predictions across the entire parameter range without the need for exhaustive Monte Carlo runs.

### Implementation

The feature is implemented as a modular package within the experimental directory to maintain core library stability while allowing for iterative community feedback:

1) `utils.py`: Implements `batch_to_xy`, a utility to transform the `list[dict]` output from `batch_run` into feature matrices (X) and target vectors (y) compatible with `scikit-learn`.

2) `sampling.py`: Provides `sample_parameters` using Latin Hypercube Sampling (LHS) via `scipy.stats.qmc` to ensure optimal coverage of the parameter space with minimal samples.

3) `emulator.py`: Contains the Emulator class, a high-level interface that manages the training of Scikit-learn models and provides a `.predict()` method for fast approximation.

4) `tests/experimental/test_surrogate.py`: A new test suite verifying the accuracy of data transformation, sampling bounds, and the end-to-end training workflow.

### Usage Examples

The following example demonstrates training an emulator for the Boltzmann Wealth Model to predict the Gini coefficient:

```python
from mesa.examples.basic.boltzmann_wealth_model.model import BoltzmannWealth
from mesa.experimental.surrogate import Emulator, sample_parameters

param_space = {"width": (10, 50), "height": (10, 50), "n": (50, 200)}
samples = sample_parameters(param_space, n_samples=20)

emu = Emulator(BoltzmannWealth, ["width", "height", "n"])
emu.train(samples, output_metric="Gini")

prediction = emu.predict({"width": 25, "height": 25, "n": 120})
print(f"Predicted Gini Coefficient: {prediction:.4f}")
```

```
Predicted Gini Coefficient: 0.6159
```

### Additional Notes

- This module introduces optional dependencies on `scipy` (for sampling) and `scikit-learn` (for the default regressor).

- The Emulator class requires the model to have a `DataCollector` initialized to capture the `output_metric` during training runs.

- All files have been updated to comply with Ruff standards, including full PEP 257 docstring coverage and PEP 8 naming conventions.

### CODE REVIEW

**Comment by quaquel** [comment]
At the moment, batchrun and datacollection are about to be removed as part of the move to Mesa 4.0. Moreover, as discussed in the [Mesa 4 topic](https://github.com/mesa/mesa/discussions/2972), we have to be careful to cleanly delineate the scope of Mesa. In my view, surrogate modelling is outside the scope for Mesa itself. The focus really is on being the best possible Python library for agent-based modeling while offloading other concerns (e.g., calibration, sensitivity analysis, surrogate modeling) to external libraries. So, I am closing this PR.

However, I encourage you to turn it into an example for mesa-examples to showcase how you can connect scikit with mesa for surrogate modelling.
